cos-fleetshard-support/src/main/java/org/bf2/cos/fleetshard/support/client/EventClient.java

```diff
@@ -0,0 +1,120 @@
+package org.bf2.cos.fleetshard.support.client;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Optional;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.Event;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectReference;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.utils.KubernetesResourceUtil;
+
+@ApplicationScoped
+public class EventClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventClient.class);
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    public enum EventType {
+        NORMAL("Normal"),
+        WARNING("Warning");
+
+        private String text;
+
+        EventType(String text) {
+            this.text = text;
+        }
+
+        public String getText() {
+            return text;
+        }
+
+        public static Optional<EventType> fromText(String text) {
+            return Arrays.stream(values()).filter(e -> e.getText().equals(text)).findFirst();
```

Comment:
This is not going to be an issue given that this method is probably not invoked many time, however this produces a number of temporary objects, i.e:
- invoking `values()` generates a new array every time it is invoked and such array it is often cached as a static field
- it allocates a stream and the related internal objects

Most of such temporary allocation can be avoided by something like:

```java
public enum EventType {
    static final EventType[] VALS = values();
    
    public static Optional<EventType> fromText(String text) {
        for (int i = 0; i < val.lenght; i++} {
            if (VALS[i].getText().equals(text)) {
                return Optional.of(VALS[i]);
            }
            
            return Optional.empty();
        }
    }
}
```